src/nbl/ext/MitsubaLoader/CMitsubaMaterialCompilerFrontend.cpp

```diff
@@ -92,19 +92,34 @@ CMitsubaMaterialCompilerFrontend::EmitterNode* CMitsubaMaterialCompilerFrontend:
         {
             profile.texture = { image, sampler, 1.f };
 
-            const bool inFlattenDomain = _emitter->area.emissionProfile->flatten >= 0.0 && _emitter->area.emissionProfile->flatten <= 1.0;
+            const bool inFlattenDomain = _emitter->area.emissionProfile->flatten >= 0.0 && _emitter->area.emissionProfile->flatten < 1.0; // [0, 1)
 
-            if (inFlattenDomain)
+            if (!inFlattenDomain)
             {
-                auto flattenIES = meta->profile.createIESTexture(m_loaderContext->override_, _emitter->area.emissionProfile->filename, _emitter->area.emissionProfile->flatten);
+                const auto cFlatten = core::min(core::abs(_emitter->area.emissionProfile->flatten), 1.f - std::numeric_limits<float>::epsilon());
+                os::Printer::log("WARNING: Flatten property = " + std::to_string(_emitter->area.emissionProfile->flatten) + " is outside it's [0, 1) domain - the flatten property will be clamped to " + std::to_string(cFlatten), ELL_WARNING);
+                _emitter->area.emissionProfile->flatten = cFlatten;
+            }
```

Comment:
just do
```cpp
flatten = abs(flatten);
if (flatten>1.f)
{
   flatten = 1.f;
   error
}
else if (flatten>1.f-epsilon)
   warning
```

Reply:
```c++
else if (flatten>1.f-epsilon)
   warning
```

why would we allow for `flatten=1.f`?

Reply:
1.f should be error imo not a warning, we should not let it pass should we?

Reply:
> ```c++
> else if (flatten>1.f-epsilon)
>    warning
> ```
> 
> why would we allow for `flatten=1.f`?

implied domain flatten mode, with a conical profile, makes it a uniform cone wheras the lack of an IES profile would make it a sphere or hemisphere

Reply:
need a separate warning and error in next PR

Reply:
> need a separate warning and error in next PR

https://github.com/Devsh-Graphics-Programming/Nabla/commit/1b5131dbe039f59c7a0011d52a9a2a44757f87ae